src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,371 @@
+        )
+        assert (
+            customer_detail.address
+            == payload["data"]["object"]["customer_details"]["address"]["line1"]
+        )
+        assert (
+            customer_detail.postal_code
+            == payload["data"]["object"]["customer_details"]["address"]["postal_code"]
+        )
+        assert (
+            customer_detail.email
+            == payload["data"]["object"]["customer_details"]["email"]
+        )
+
+        subscription = Subscription.objects.get(user=user)
+        assert subscription.stripe_subscription_id == payload["data"]["object"]["id"]
+        assert subscription.status == payload["data"]["object"]["status"]
+        assert (
+            subscription.payment_status == payload["data"]["object"]["payment_status"]
+        )
+        assert (
+            subscription.product_name
+            == payload["data"]["object"]["metadata"]["product_name"]
+        )
+        assert subscription.total_price == payload["data"]["object"][
+            "amount_total"
+        ] / Decimal(100)
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_checkout_session_completed_user_does_not_exist(
+        self, mock_construct_event, client, payload
+    ):
+        mock_construct_event.return_value = payload
+        payload["data"]["object"]["metadata"]["user_id"] = "9999"
+
+        sig_header = generate_stripe_signature(payload, STRIPE_WEBHOOK_SECRET)
+        response = client.post(
+            reverse_lazy("v1:payment:stripe-webhook"),
+            data=json.dumps(payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,
+        )
+        assert response.status_code == 400
+        # assert response.content == b"User does not exist" ToDo
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_customer_subscription_updated(self, mock_construct_event, client, user):
+        """
+        Complete test of the `customer.subscription.updated` function.
+        Checks that `Subscription` is correctly updated.
+        """
+        # Test data simulating the Stripe event
+        payload = {
+            "type": "customer.subscription.updated",
+            "data": {
+                "object": {
+                    "current_period_end": 1734440765,  # Timestamp for period end
+                    "current_period_start": 1731848765,  # Timestamp for period start
+                    "customer": "cus_REbNQXKKFCRF2c",
+                    "cancel_at_period_end": False,
+                    "trial_end": "null",  # No trial
+                    "items": {
+                        "data": [
+                            {
+                                "plan": {
+                                    "active": True,  # Plan status
+                                },
+                            },
+                        ],
+                    },
+                }
+            },
+        }
```

Comment:
Maybe, move up next to the other payload (checkout_session_completed_payload())?

Reply:
I've renamed it to keep consistency and clarity in the code